docs/Middleware-and-Component.md

```diff
@@ -60,6 +60,49 @@ $middlewareStack->add(new AuthorizationMiddleware($this, [
 ]));
 ```
 
+### Handling unauthorized requests
+
+By default authorization exceptions thrown by the application are rethrown by the middleware.
+You can configure handlers for unauthorized requests and perform custom action, e.g.
+redirect the user to the login page.
+
+The built-in handlers are:
+
+* `Exception` - this handler will rethrow the exception, this is a default behavior of the middleware.
+* `Redirect` - this handler will redirect the request to the provided URL.
+* `CakeRedirect` - redirect handler with support for CakePHP Router.
+
+Both redirect handlers share the same configuration options:
+
+* `url` - URL to redirect to (`CakeRedirect` supports array syntax).
+* `exceptions` - a list of exception classes that should be redirected. By default only `MissingIdentityException` is redirected.
+* `queryParam` - the accessed request URL will be attached to the redirect URL query parameter (`url` by default).
+* `statusCode` - HTTP status code of a redirect, `302` by default.
+
+For example:
+
+```php
+$middlewareStack->add(new AuthorizationMiddleware($this, [
+    'unauthorizedHandler' => [
+        'className' => 'Authorization.Redirect',
```

Comment:
Why not calling them what they are? StringUrlRedirect vs ArrayUrlRedirect maybe?
The CakeRedirect doesnt really say what it does (it only supports array, nothing else from what I read).

Reply:
Cake as a name suggests supports CakePHP :).

Reply:
I don't get that - isn't everything here something for "Cake"?
After all:
```
"require": {
		"cakephp/core": "~3.5",
```

So clarying what those Handlers actually use seems to be more important than outlining that one is for Cake, and the other somewhat too :)

Reply:
> isn't everything here something for "Cake"?

@dereuromark No, it is a framework-agnostic plugin.